sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs

```diff
@@ -3388,6 +3533,54 @@ public void UnexpectedException(string errorMessage)
             }
         }
 
+        /// <summary>
+        ///   Writes an event with four string arguments and two value type arguments into a stack allocated
+        ///   <see cref="EventSource.EventData"/> struct to avoid the parameter array allocation on the WriteEvent methods.
+        /// </summary>
+        ///
+        /// <param name="eventId">The identifier of the event.</param>
+        /// <param name="arg1">The first argument.</param>
+        /// <param name="arg2">The second argument.</param>
+        /// <param name="arg3">The third argument.</param>
+        /// <param name="arg4">The fourth argument.</param>
+        /// <param name="arg5">The fifth argument.</param>
+        ///
```

Comment:
fancy! Should we adopt this pattern in other perf critical paths such as Core?

Reply:
Alex did a prototype for core once upon a time, and the micro-benchmarks didn't show much improvement, IIRC.    Struck me as odd, considering that without these you fall into a general `WriteEvent(params object[] args)` overload and pay for allocating the array.   The approach in the messaging libraries had a non-trivial positive impact.

https://github.com/Azure/azure-sdk-for-net/pull/33913